src/typedefs.ts

```diff
@@ -0,0 +1,11 @@
+interface NominalTag<T> {
+  'nominalTag': T;
```

Comment:
```suggestion
  'nominalTag'?: T;
```

https://github.com/fabricjs/fabric.js/pull/8019#discussion_r903596787